### PR TITLE
WIP - don't merge - Making Courtfinder app use company-wide docker images with pre-installed dependencies

### PR DESCRIPTION
**Pull request by lukaszraczylo**
Making Courtfinder app use company-wide docker images with pre-installed dependencies and configuration together with Dockerfile miniaturization and standardization across projects.
- created minimal Dockerfile 
- cleanup after 'old' Dockerized version which isn't needed anymore.
- moved test and development related entries from Gemfile into appropriate groups

### CODE REVIEW

**Comment by lukaszraczylo** [comment]
PR awaiting merge of https://github.com/ministryofjustice/docker-templates/pull/20 which it requires to build project.

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    bundle exec rake db:migrate
+    ;;
+create)
+    echo "running seed"
+    bundle exec rake db:create
+    bundle exec rake db:migrate
+    bundle exec rake db:seed
+    ;;
+esac
+bundle exec rails s
```
Comment (mattmb) [inline run.sh]: As discussed, let's ditch runit and switch this line to:
`bundle exec unicorn -p $UNICORN_PORT -c ./config/unicorn.rb`

Also check that the config/unicorn.rb exists in the app and at least sets the number of workers.

**Comment by mattmb** [comment]
So I don't quite get what's going on with the assets...

Starting the app in development mode they all get served correctly, but something seems to be broken in production mode?

File: run.sh (hunk 1 above)
Comment (lukaszraczylo) [inline run.sh]: Unicorn gem is installed via bundler but never used except of testing. As it's "just" admin panel it's safe to assume that load won't be that massive and it's safe to use built-in server. Just my opinion :)

**Comment by lukaszraczylo** [comment]
Issue with assets:
- they're present in old version of the app, in its S3 container.
- they were not present in dockerised version of the app.
I assumed they were leftovers from times when those assets were actually created during deployment process and because app used the same container all the time - they were there, not touched, not changed.

**Comment by mattmb** [comment]
So it seems to have been pulling them from https://assets.digital.cabinet-office.gov.uk which could explain it. My only reservation is that just including the static files might be a pain if they were compiled from something in the past.

**Comment by lukaszraczylo** [comment]
Agree - but luckily files present in the assets folder can/will be overwritten during asset generation ( if something generates them ), so it's a win-win situation IMO.

**Pull request closed without merging**